plugins/analytics/public/virustotal_api.py

```diff
@@ -80,57 +82,66 @@ def analyze(observable, results):
+                        o_url = Url.get_or_create(value=detected_url['url'])
+                        links.update(
+                            o_url.active_link_to(
+                                o_url, 'hostname', 'virustotal_query'))
 
         elif isinstance(observable, Hostname):
-            if json_result.get('permalink'):
-                result['permalink'] = json_result['permalink']
+            if json_result['response_code'] == 1:
+                if json_result.get('permalink'):
+                    result['permalink'] = json_result['permalink']
 
-            result['positives'] = json_result.get('positives', 0)
+                result['positives'] = json_result.get('positives', 0)
 
-            if json_result.get('total'):
-                result['total'] = json_result['total']
+                if json_result.get('total'):
+                    result['total'] = json_result['total']
 
         elif isinstance(observable, Hash):
 
-            result['positives'] = json_result['positives']
+            if json_result['response_code'] == 1:
 
-            if 'permalink' in json_result:
-                result['permalink'] = json_result['permalink']
+                result['positives'] = json_result.get['positives']
 
-            if 'total' in json_result:
-                result['total'] = json_result['total']
+                if 'permalink' in json_result:
+                    result['permalink'] = json_result['permalink']
 
-            hashes = {
-                'md5': json_result['md5'],
-                'sha1': json_result['sha1'],
-                'sha256': json_result['sha256']
-            }
-            create_hashes = [
-                (k, v) for k, v in hashes.items() if v != observable.value
-            ]
+                if 'total' in json_result:
+                    result['total'] = json_result['total']
 
-            for k, v in create_hashes:
-                new_hash = Hash.get_or_create(value=v)
-                new_hash.tag(observable.get_tags())
-                links.update(
-                    new_hash.active_link_to(observable, k, 'virustotal_query'))
+                hashes = {
+                    'md5': json_result['md5'],
+                    'sha1': json_result['sha1'],
+                    'sha256': json_result['sha256']
+                }
+                create_hashes = [
+                    (k, v) for k, v in hashes.items() if v != observable.value
+                ]
 
+                for k, v in create_hashes:
+                    new_hash = Hash.get_or_create(value=v)
+                    new_hash.tag(observable.get_tags())
+                    links.update(
+                        new_hash.active_link_to(observable, k,
+                                                'virustotal_query'))
+                result['scan_date'] = json_result['scan_date']
+            else:
+                result['scan_date'] = datetime.utcnow().strftime(
+                    '%Y-%m-%d %H:%M:%S')

```

Comment:
I assume that `'%Y-%m-%d %H:%M:%S'` is the same format as `json_result['scan_date']`?

Reply:
Yep the same :)